File: mobile/android/OpenU60/app/src/main/java/com/openu60/core/network/AgentError.kt
package com.openu60.core.network

sealed class AgentError(override val message: String, override val cause: Throwable? = null) : Exception(message, cause) {
    class Unauthorized : AgentError("未认证，请先登录。")
    class ServerError(msg: String) : AgentError("服务器错误：$msg")
    class NetworkError(msg: String, cause: Throwable? = null) : AgentError("网络错误：$msg", cause)
    class DecodingError(msg: String) : AgentError("解析响应失败：$msg")
    class ServerUnreachable : AgentError("无法连接到代理")
    class Timeout : AgentError("请求超时")
}

File: mobile/ios/OpenU60/Core/Networking/AgentError.swift
import Foundation

enum AgentError: LocalizedError {
    case unauthorized
    case serverError(String)
    case networkError(Error)
    case decodingError(String)
    case serverUnreachable
    case timeout

    var errorDescription: String? {
        switch self {
        case .unauthorized:
            return "未认证，请先登录。"
        case .serverError(let message):
            return "服务器错误：\(message)"
        case .networkError(let error):
            return "网络错误：\(error.localizedDescription)"
        case .decodingError(let detail):
            return "解析响应失败：\(detail)"
        case .serverUnreachable:
            return "无法连接到代理"
        case .timeout:
            return "请求超时"
        }
    }
}
